web/forms/implementing_simple_autocompletion/my_element.dart

```diff
@@ -1,97 +1,120 @@
+@PolymerRegister('my-element')
 class MyElement extends PolymerElement {
-
   /// In a real project, the following would live elsewhere, e.g. a backend.
-  static const List<String> hardCodedListOfWords = const [
-    "apple", "ananas", "banana", "beetroot", "cassava", "cherry", "cucumber",
-    "eggplant", "grapes", "tomato", "pear", "pomegranate", "potato",
-    "lettuce", "strawberry"
+  static const List<String> _hardCodedListOfWords = const [
+    "apple",
+    "ananas",
+    "banana",
+    "beetroot",
+    "cassava",
+    "cherry",
+    "cucumber",
+    "eggplant",
+    "grapes",
+    "tomato",
+    "pear",
+    "pomegranate",
+    "potato",
+    "lettuce",
+    "strawberry"
   ];
 
-  @observable String inputQuery;
-  @observable final List<String> matchingCandidates = toObservable([]);
-  @observable int selectedItemIndex = -1;
-  @observable String selectedCandidate;
+  @property String inputQuery;
+  @property final List<String> matchingCandidates = [];
+  @property int selectedItemIndex = -1;
+  @property String selectedCandidate;
 
   MyElement.created() : super.created() {}
 
-  void keyDown(Event event, Object object, CoreInput target) {
+  @reflectable
+  void keyDown(Event event, [_]) {
     if (!(event is KeyboardEvent)) {
       return;
     }
     switch ((event as KeyboardEvent).keyCode) {
       case KeyCode.DOWN:
-        moveSelection(1);
+        _moveSelection(1);
         event.preventDefault();
         break;
       case KeyCode.UP:
-        moveSelection(-1);
+        _moveSelection(-1);
         event.preventDefault();
         break;
       case KeyCode.ENTER:
         if (selectedItemIndex != -1) {
           window.alert("You picked ${selectedCandidate} from the list.");
         } else {
-          window.alert("You hit enter on ${inputQuery} without selecting it from the list.");
+          window.alert(
+              "You hit enter on ${inputQuery} without selecting it from the list.");
         }
         event.preventDefault();
         break;
     }
   }
 
-  void moveSelection(int delta) {
-    selectedItemIndex += delta;
+  void _moveSelection(int delta) {
+    set('selectedItemIndex', selectedItemIndex + delta);
     if (selectedItemIndex < 0) {
       selectedItemIndex = 0;
     }
     if (selectedItemIndex >= matchingCandidates.length) {
       selectedItemIndex = matchingCandidates.length - 1;
     }
-    if (0 <= selectedItemIndex && selectedItemIndex < matchingCandidates.length) {
-      selectedCandidate = matchingCandidates[selectedItemIndex];
+    if (0 <= selectedItemIndex &&
+        selectedItemIndex < matchingCandidates.length) {
+      set('selectedCandidate', matchingCandidates[selectedItemIndex]);
     } else {
-      selectedCandidate = "";
+      set('selectedCandidate', '');
     }
   }
 
-  void findCandidates(Event event, Object object, CoreInput target) {
-    matchingCandidates.clear();
-    inputQuery = target.value;
+  @reflectable
+  String selectedClass(String candidate) =>
```

Comment:
per change in the html, add a second argument here, either just `_` or `String selectedCandidate`.